Composer/packages/client/src/router.tsx

```diff
@@ -77,7 +77,7 @@ const ProjectRouter: React.FC<RouteComponentProps<{ projectId: string }>> = prop
     if (state.projectId !== props.projectId && props.projectId) {
       actions.fetchProjectById(props.projectId);
     }
-  }, [props.projectId, state.projectId]);
+  }, [props.projectId]);
```

Comment:
After deleting the bot, the state.projectId will set to ''. It changed so  actions.fetchProjectById(props.projectId). But the bot {props.projectId} is already deleted. 

I think the logic is, if the url changed, props.projectId will change, so we fetch the project. The we successfully fetched the project, then change store projectId.

Reply:
This isn't a criticism on this change, but a general question for people who understand more about this design here than I do. If the intended logic for the useEffect doesn't involve `state.projectId`, can we simplify line 77 to just `if (props.projectId !== '')`? Why do we care about the state inside that useEffect call?